src/preFilter.ts

```diff
@@ -18,14 +19,18 @@ export function filterByContent(searchee: Searchee): boolean {
 		});
 	}
 
-	const isSingleEpisodeTorrent =
-		searchee.files.length === 1 && EP_REGEX.test(searchee.files[0].name);
-
-	if (!includeEpisodes && isSingleEpisodeTorrent) {
+	const isSingleEpisodeTorrent = searchee.files.length === 1 && EP_REGEX.test(searchee.name); 
+	const isSeasonPack = typeof searchee.path !== 'undefined' ? SEASON_REGEX.test(path.dirname(searchee.path).split(path.sep).reverse()[0]): false;
```

Comment:
checking for undefined via `typeof` is the most correct but a bit formal. For the purposes of this check we can probably just check that it's truthy

Additionally, we can use `basename` to our advantage here:

```ts
const comesFromSeasonPack = searchee.path &&
	SEASON_REGEX.test(basename(dirname(searchee.path))
```

Reply:
Correct me if I'm wrong, but after some thought would it not make sense to save some CPU cycles and eliminate searchee.path's truthy check altogether?

Assuming the SEASON_REGEX.test returns false when it's provided an undefined value (I'd expect this behavior) - this actually seems redundant being in the same initialization.

`const isSeasonPackEpisode = SEASON_REGEX.test(path.basename(path.dirname(searchee.path)));`

Reply:
> `SEASON_REGEX.test(basename(dirname(searchee.path))`

This is not correct. For a season pack episode, `searchee.path` will be /torrents/tv/Show.S01/, dirname will give `torrents/tv`, basename gives `Show.S01`, basename(dirname) gives `tv`. So just basename should be the way to go.

> would it not make sense to save some CPU cycles and eliminate searchee.path's truthy check altogether?

If it's undefined, it should do the same amount of processing on both due to short circuiting. If the left side of the && is false, it not try to evaluate the right side. `.test()` does appear to handle undefined though, so maybe the extra null check that is done if it's true is not worth it, but I prefer the hard null check before running `test`

Reply:
I want to be explicit that searchee.path has to exist. Otherwise we're relying on behavior of dirname and basename when it encounters undefined, which feels bad.

Reply:
> If it's undefined, it should do the same amount of processing on both due to short circuiting. If the left side of the && is false, it not try to evaluate the right side. .test() does appear to handle undefined though, so maybe the extra null check that is done if it's true is not worth it, but I prefer the hard null check before running test

Noted, will keep the ternary as is with the SEASON_REGEX i commented previous your comment.

`const isSeasonPackEpisode = searchee.path && SEASON_REGEX.test(path.basename(path.dirname(searchee.path)));`

Functions as intended.

Reply:
nit, but I'd prefer `&&` to using a ternary here

Reply:
> nit, but I'd prefer `&&` to using a ternary here

Sure, makes it more readable anyway. Will use the &&.